pkg/app/pipedv1/plugin/kubernetes/deployment/server.go

```diff
@@ -175,3 +181,89 @@ func (a *DeploymentService) loadManifests(ctx context.Context, deploy *model.Dep
 
 	return manifests, nil
 }
+
+func (a *DeploymentService) ExecuteStage(ctx context.Context, request *deployment.ExecuteStageRequest) (*deployment.ExecuteStageResponse, error) {
+	switch request.GetInput().GetStage().GetName() {
+	case StageK8sSync.String():
+		return a.executeK8sSyncStage(ctx, request.GetInput())
+	case StageK8sRollback.String():
+		return a.executeK8sRollbackStage(ctx, request.GetInput())
+	default:
+		return nil, status.Error(codes.InvalidArgument, "unimplemented or unsupported stage")
+	}
+}
+
+func (a *DeploymentService) executeK8sSyncStage(ctx context.Context, input *deployment.ExecutePluginInput) (*deployment.ExecuteStageResponse, error) {
+	lp := a.LogPersister.StageLogPersister(input.GetDeployment().GetId(), input.GetStage().GetId())
+	lp.Infof("Start syncing the deployment")
+
+	cfg, err := config.DecodeYAML[*kubeconfig.KubernetesApplicationSpec](input.GetTargetDeploymentSource().GetApplicationConfig())
+	if err != nil {
+		return nil, status.Error(codes.InvalidArgument, err.Error())
+	}
+
+	lp.Infof("Loading manifests at commit %s for handling", input.GetDeployment().GetTrigger().GetCommit().GetHash())
+	manifests, err := a.loadManifests(ctx, input.GetDeployment(), cfg.Spec, input.GetTargetDeploymentSource())
+	if err != nil {
+		lp.Errorf("Failed while loading manifests (%v)", err)
+		return nil, status.Error(codes.Internal, err.Error())
+	}
+	lp.Successf("Successfully loaded %d manifests", len(manifests))
+
+	// Because the loaded manifests are read-only
+	// we duplicate them to avoid updating the shared manifests data in cache.
+	// TODO: implement duplicateManifests function
+
+	// When addVariantLabelToSelector is true, ensure that all workloads
+	// have the variant label in their selector.
+	var (
+		variantLabel   = cfg.Spec.VariantLabel.Key
+		primaryVariant = cfg.Spec.VariantLabel.PrimaryValue
+	)
+	// TODO: handle cfg.Spec.QuickSync.AddVariantLabelToSelector
+
+	// Add builtin annotations for tracking application live state.
+	addBuiltinAnnotations(
+		manifests,
+		variantLabel,
+		primaryVariant,
+		input.GetDeployment().GetTrigger().GetCommit().GetHash(),
+		input.GetDeployment().GetPipedId(),
+		input.GetDeployment().GetApplicationId(),
```

Comment:
For these annotations, how about adding it under loadManifests logic? This is quite common and should be force-added to any manifests under pipecd management.

Reply:
It's nice. I'll change the codes. Thanks.

Reply:
@khanhtc1202 
fixed on this commit. Please check it.
https://github.com/pipe-cd/pipecd/pull/5371/commits/390bcabf79f3001ed6e9bf3c9425d510255f427b